sdks/java/core/pom.xml

```diff
@@ -430,6 +430,19 @@
 
     <!-- test dependencies -->
     <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>beam-sdks-common-fn-api</artifactId>
+      <type>test-jar</type>
```

Comment:
We chatted and we are on the same page about where we want to be, so I am cool with putting this in a slightly different place for the time being.